components/eventing-controller/controllers/subscription/nats/reconciler.go

```diff
@@ -193,38 +190,16 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 
 	if !subscription.ObjectMeta.DeletionTimestamp.IsZero() {
 		// The object is being deleted
-		if utils.ContainsString(subscription.ObjectMeta.Finalizers, Finalizer) {
-			if err := r.Backend.DeleteSubscription(subscription); err != nil {
-				log.Errorw("delete subscription failed", "error", err)
-				// if failed to delete the external dependency here, return with error
-				// so that it can be retried
-				return ctrl.Result{}, err
-			}
-
-			// remove our finalizer from the list and update it.
-			subscription.ObjectMeta.Finalizers = utils.RemoveString(subscription.ObjectMeta.Finalizers, Finalizer)
-			if err := r.Client.Update(ctx, subscription); err != nil {
-				events.Warn(r.recorder, subscription, events.ReasonUpdateFailed, "Update Subscription failed %s", subscription.Name)
-				log.Errorw("remove finalizer from subscription failed", "error", err)
-				return checkIsConflict(err)
-			}
-			log.Debug("remove finalizer from subscription succeeded")
-		}
-		// Stop reconciliation as the object is being deleted
-		return ctrl.Result{}, nil
+		err := r.handleSubscriptionDeletion(ctx, subscription, log)
+		return ctrl.Result{}, err
```

Comment:
IMO it's harder to read such oneliners

Reply:
@mfaizanse  `handleSubscriptionDeletion` only returns an `error`. We need `ctrl.Result` and `error` though

Reply:
I mean you can change the return type of `handleSubscriptionDeletion` to `(ctrl.Result, error)`.

Reply:
Yes I could do that, but I think we could do it in this way two, because
- it's only one additional line
- imagine if you would want to add something after the deletion, i.e. an error log, in that case you don't need to return right away
- we already do it both ways in our code

So I would keep that.